### PR TITLE
Update cluster-role.yaml

### PR DESCRIPTION
**Pull request by xh3b4sd**
We need to extend the RBAC rules in order to run the jobs for `awscnfm` on the control planes. I am not sure how open we want to be with that. I can see that the test infra just needs access to everything. 

```
[test-aws : run-tests] panic: {"kind":"unknown","annotation":"clusters.cluster.x-k8s.io is forbidden: User \"system:serviceaccount:giantswarm:test-infra\" cannot create resource \"clusters\" in API group \"cluster.x-k8s.io\" in the namespace \"default\"","stack":[{"file":"/root/project/cmd/action/create/cluster/defaultcontrolplane/runner.go","line":79},{"file":"/root/project/cmd/action/create/cluster/defaultcontrolplane/runner.go","line":32},{"file":"/root/project/pkg/plan/executor.go","line":84},{"file":"/go/pkg/mod/github.com/giantswarm/backoff@v0.2.0/retry.go","line":13},{"file":"/root/project/pkg/plan/executor.go","line":92},{"file":"/root/project/cmd/plan/pl001/runner.go","line":63},{"file":"/root/project/cmd/plan/pl001/runner.go","line":31},{"file":"/root/project/main.go","line":78}]}
```

### CODE REVIEW

File: control-plane/cluster-role.yaml [hunk 1]
```diff
@@ -4,8 +4,16 @@ metadata:
...
 rules:
   - apiGroups:
+      - cluster.x-k8s.io
+      - infrastructure.giantswarm.io
       - release.giantswarm.io
     resources:
       - "*"
     verbs:
       - "*"
+  - apiGroups:
```
Comment (MarcelMue) [inline control-plane/cluster-role.yaml:14]: Why is this needed?

**Comment by MarcelMue** [comment]
@yulianedyalkova How are changes to the different CPs rolled out? As this needs to take effect on `gaia`,`godsmack` and `gorgoth` afaik

File: control-plane/cluster-role.yaml (hunk 1 above)
Comment (xh3b4sd) [reply control-plane/cluster-role.yaml:14]: My understanding was that this is for e.g. pods and services. All the k8s native runtime objects. We have this in the `aws-operator` as well.

**Review by calvix — APPROVED** [review]

**Comment by xh3b4sd** [comment]
The thing with this is that I am not sure this helped fixing our problems. Because in the end we had to manually modify the cluster role on `gaia` for the `release-manager`. That is like a real maze. This one here can probably be closed and as soon as someone does something on `gaia` it will probably break without the system healing itself.

**Comment by MarcelMue** [comment]
There is a tracking issue here to deal with the current issues including automatic propagation:
https://github.com/giantswarm/giantswarm/issues/13569

**Comment by yulianedyalkova** [comment]
> The thing with this is that I am not sure this helped fixing our problems. Because in the end we had to manually modify the cluster role on `gaia` for the `release-manager`. That is like a real maze. This one here can probably be closed and as soon as someone does something on `gaia` it will probably break without the system healing itself.

Even if there is currently no automatic way to apply the changes in the `control-plane` folder, it should still represent the current state of those objects on the installations. So you should update the PR with the changes you did manually on gaia.

**Comment by xh3b4sd** [comment]
We just used the changes from here. So then we should just merge this PR.
